File: Leetcode/Math/#326-Power of Three/main.cpp
class Solution {
public:
    bool isPowerOfThree(int n) {
        if (n <= 0) {
            return false;
        }

        if (n == 1) {
            return true;
        }

        if (n % 3 != 0) {
            return false;
        }

        return isPowerOfThree(n / 3);
    }
};

File: Leetcode/Math/#326-Power of Three/main.py
class Solution(object):
    def isPowerOfThree(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n < 3:
            if n != 1: return False
            else: return True
        elif n % 3 == 0: return self.isPowerOfThree(n / 3)
        else: return False
